validator/client/src/test/java/tech/pegasys/teku/validator/client/EpochDutiesTest.java

```diff
@@ -35,136 +35,162 @@
 
 class EpochDutiesTest {
 
-  private static final UInt64 EPOCH = UInt64.valueOf(10);
-  private final SafeFuture<Optional<ScheduledDuties>> scheduledDutiesFuture = new SafeFuture<>();
-  private final DutyLoader dutyLoader = mock(DutyLoader.class);
-  private final ScheduledDuties scheduledDuties = mock(ScheduledDuties.class);
-  private final Optional<ScheduledDuties> scheduledDutiesOptional = Optional.of(scheduledDuties);
+  protected static final UInt64 EPOCH = UInt64.valueOf(10);
+  protected final SafeFuture<Optional<ScheduledDuties<?, ?>>> scheduledDutiesFuture =
+      new SafeFuture<>();
 
-  private EpochDuties duties;
+  protected final DutyLoader dutyLoader = mock(DutyLoader.class);
+  protected final ScheduledDuties<?, ?> scheduledDuties = mock(ScheduledDuties.class);
+
+  protected final Optional<ScheduledDuties<?, ?>> scheduledDutiesOptional =
+      Optional.of(scheduledDuties);
+
+  protected EpochDuties duties;
 
   @BeforeEach
   void setUp() {
     when(dutyLoader.loadDutiesForEpoch(EPOCH)).thenReturn(scheduledDutiesFuture);
-    duties = EpochDuties.calculateDuties(dutyLoader, EPOCH);
+    duties = calculateDuties(dutyLoader, EPOCH);
     verify(dutyLoader).loadDutiesForEpoch(EPOCH);
   }
 
-  @Test
-  void cancel_shouldCancelFuture() {
-    duties.cancel();
-    assertThat(scheduledDutiesFuture).isCancelled();
+  private EpochDuties calculateDuties(final DutyLoader dutyLoader, final UInt64 epoch) {
```

Comment:
(nit) Doesn't look like this helper method is doing much

Reply:
True. It was at one point an abstract method because `EpochDuties` had to have a subclass for blocks vs attestations. Removed now.